airflow/providers/teradata/transfers/azure_blob_to_teradata.py

```diff
@@ -0,0 +1,105 @@
+
+from airflow.models import BaseOperator
+from airflow.providers.microsoft.azure.hooks.wasb import WasbHook
+from airflow.providers.teradata.hooks.teradata import TeradataHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class AzureBlobStorageToTeradataOperator(BaseOperator):
+    """
+
+    Loads CSV, JSON and Parquet format data from Azure Blob Storage to Teradata.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AzureBlobStorageToTeradataOperator`
+
+    :param blob_source_key: The URI format specifying the location of the Azure blob object store.(templated)
+        The URI format is `/az/YOUR-STORAGE-ACCOUNT.blob.core.windows.net/YOUR-CONTAINER/YOUR-BLOB-LOCATION`.
+        Refer to
+        https://docs.teradata.com/search/documents?query=native+object+store&sort=last_update&virtual-field=title_only&content-lang=en-US
+    :param azure_conn_id: The Airflow WASB connection used for azure blob credentials.
+    :param teradata_table: The name of the teradata table to which the data is transferred.(templated)
+    :param teradata_conn_id: The connection ID used to connect to Teradata
+        :ref:`Teradata connection <howto/connection:Teradata>`
+
+    Note that ``blob_source_key`` and ``teradata_table`` are
+    templated, so you can use variables in them if you wish.
+    """
+
+    template_fields: Sequence[str] = ("blob_source_key", "teradata_table")
+    ui_color = "#e07c24"
+
+    def __init__(
+        self,
+        *,
+        blob_source_key: str,
+        azure_conn_id: str = "azure_default",
+        teradata_table: str,
+        teradata_conn_id: str = "teradata_default",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.blob_source_key = blob_source_key
+        self.azure_conn_id = azure_conn_id
+        self.teradata_table = teradata_table
+        self.teradata_conn_id = teradata_conn_id
+
+    def execute(self, context: Context) -> None:
+        self.log.info(
+            "transferring data from %s to teradata table %s...", self.blob_source_key, self.teradata_table
+        )
+        azure_hook = WasbHook(wasb_conn_id=self.azure_conn_id)
+        conn = azure_hook.get_connection(self.azure_conn_id)
+        # Obtaining the Azure client ID and Azure secret in order to access a specified Blob container
+        access_id = conn.login if conn.login is not None else ""
+        access_secret = conn.password if conn.password is not None else ""
+        teradata_hook = TeradataHook(teradata_conn_id=self.teradata_conn_id)
+        sql = f"""
+                    CREATE MULTISET TABLE {self.teradata_table}  AS
+                    (
+                        SELECT * FROM (
+                            LOCATION = '{self.blob_source_key}'
+                            ACCESS_ID= '{access_id}'
+                            ACCESS_KEY= '{access_secret}'
```

Comment:
Won't this expose the secret in the logs?

Reply:
![Access_Id_Key_Log](https://github.com/apache/airflow/assets/23031170/a28a4285-0799-4485-8993-0fc9b9478414)
secret is logging as masked in the log.